crates/ruff_python_formatter/src/other/interpolated_string_element.rs

```diff
@@ -283,12 +258,31 @@ impl Format<PyFormatContext<'_>> for FormatInterpolatedElement<'_> {
             {
                 let mut f = WithNodeLevel::new(NodeLevel::ParenthesizedExpression, f);
 
-                if self.context.can_contain_line_breaks() {
-                    group(&format_args![
-                        open_parenthesis_comments,
-                        soft_block_indent(&item)
-                    ])
-                    .fmt(&mut f)?;
+                if self.context.is_multiline() {
+                    // TODO: The `or comments.has_trailing...` can be removed once newlines in format specs are a syntax error.
+                    // This is to support the following case:
+                    // ```py
+                    // x = f"{x  !s
+                    //          :>0
+                    //          # comment 21
+                    // }"
+                    // ```
+                    if format_spec.is_none() || comments.has_trailing_own_line(self.element) {
```

Comment:
I'm guessing the example mentioned in the comment is specifically to target the `format_spec.is_none` case and not the trailing comment case. And, as the latter case is going to be removed, we don't really need to provide an example for that here.

Reply:
The example is for the `format_spec.is_some` case. It will be removed in my next PR. 

I mainly provided it because I found it helpful to reason about all the changes i've been making. I think I keep it, considering that I already have a PR that removes it again.